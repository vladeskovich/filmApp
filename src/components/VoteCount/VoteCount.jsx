import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Typography from '../Typography';
import styles from './VoteCount.scss';

const VoteCount = ({
  className,
  rating,
}) => {
  const classes = classNames(styles.voteCount, className);
  const ratingRound = Math.round((rating / 2) * 10) / 10;
  return (
    <Typography
      className={classes}
    >
      {ratingRound}
    </Typography>
  );
};

VoteCount.propTypes = {
  rating: PropTypes.number,
  className: PropTypes.string,
};

VoteCount.defaultProps = {};

export default VoteCount;
