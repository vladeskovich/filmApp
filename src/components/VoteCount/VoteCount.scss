.voteCount {
  width: 35px;
  height: 16px;
  text-align: center;
  border: 2px solid #0aaee4;
  border-radius: 4px;
  font-size: 14px;
}