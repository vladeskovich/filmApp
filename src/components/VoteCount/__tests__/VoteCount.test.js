import React from 'react';
import VoteCount from '../VoteCount';

describe('VoteCount', () => {
  const props = {
    rating: 8,
    className: 'test-vote',
  };

  it('should render VoteCount component props', () => {
    const component = mount(<VoteCount {...props}/>);

    expect(component).toMatchSnapshot();
  });
});
