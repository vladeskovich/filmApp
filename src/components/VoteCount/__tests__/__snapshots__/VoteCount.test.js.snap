// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteCount should render VoteCount component props 1`] = `
<VoteCount
  className="test-vote"
  rating={8}
>
  <Typography
    className="VoteCount__voteCount_2wL5CqNf test-vote"
    variant="body"
  >
    <p
      className="Typography__typography_1E76fqYp VoteCount__voteCount_2wL5CqNf test-vote"
    >
      4
    </p>
  </Typography>
</VoteCount>
`;
